### PR TITLE
TI-18: Comment on threads

### PR DESCRIPTION
**Pull request by noahc3**
- Comments are always sorted newest to oldest.
- Loading a thread will load up to 10 top level comments and up to 2 immediate replies to those top level comments.
  - If there are more comments on the thread, a 'Show more comments' button will appear at the end of the list.
  - If a comment has more replies not shown, a 'Show replies' or 'Show more replies' button will appear.
- Comments can be replied to endlessly to any height.
- Original poster of a comment can edit or delete their comment.
  - Deleted comments replace the username and content with `[deleted]`, it will still appear in the thread and replies will remain in-tact. Same behavior as Reddit.
- Moderators and spool owner can also delete comments.

image

### CODE REVIEW

**Review by nathanpoppe — COMMENTED** [review]
Commented 1 small thing that (I think) can be removed. 
Overall the Design/Functionality/Code is 👌

We'll have to add tests for comments at some point but looks good to merge

File: threadit-api/Controllers/v1/CommentsController.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            return Ok(comments);
+        }
+
+        [HttpGet("{threadId}/newer/{siblingCommentId}")]
+        public async Task<IActionResult> NewerComments([FromRoute] string threadId, [FromRoute] string siblingCommentId, [FromServices] CommentService commentService) {            
+            CommentFull[] comments = await commentService.NewerComments(threadId, siblingCommentId);
+            return Ok(comments);
+        }
+
+        [HttpPost("{threadId}/{parentCommentId?}")]
```
Comment (CodyWallbridge) [inline threadit-api/Controllers/v1/CommentsController.cs:39]: can we not wrap this and the first function into the same since parentCommentId is nullable?

**Comment by CodyWallbridge** [comment]
it wont let me comment on the file since it wasnt changed, but you also need to make changes in spoolRepository on line 170. When we delete a spool, you need to also delete all comments for all threads that are getting deleted. You actually may want to look at the threadRepository as well for when a thread is deleted and make sure theyre all deleted there?

File: threadit-web/src/containers/Comments/CommentFeed.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+import { useEffect, useState } from "react";
+import CommentAPI from "../../api/CommentAPI";
+import { IThreadFull } from "../../models/ThreadFull";
+import { CommentBox } from "./CommentBox";
+import { CommentItem } from "./CommentItem";
+import { CommentTree, CommentTreeNode } from "./CommentTree";
+
+export const CommentFeed = observer(({thread}: {thread: IThreadFull}) => {
+    const [commentTree, setCommentTree] = useState<CommentTree>(new CommentTree());
+    const [isLoadingReplies, setIsLoadingReplies] = useState<boolean>(false);
```
[the remark this answers is not in the document]
Comment (noahc3) [reply threadit-web/src/containers/Comments/CommentFeed.tsx:14]: This state variable is a hack to force a page rerender, and the eslint comment disables the warning to allow builds to pass.

**Review by CodyWallbridge — APPROVED** [review]

**Pull request merged**